test/pjrt/test_collective_ops_tpu.py

```diff
@@ -132,5 +138,96 @@ def test_all_to_all(self, pin_layout):
                                              list(range(world_size))]])
 
 
+# Test for collective ops from torch.distributed
+class TestDistCollectiveOpsTpu(parameterized.TestCase):
+
+  # TODO(zpcore): fix the openxla dynamo issue for dist.all_reduce
+  @staticmethod
+  def my_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+    return gm.forward
+
+  @staticmethod
+  def _all_reduce(use_dynamo: bool):
+    met.clear_all()
+
+    def _dynamo_func_wrapper(input):
+      dist.all_reduce(input, dist.ReduceOp.SUM)
+      return input
+
+    dist.init_process_group("xla", init_method='xla://')
+    device = xm.xla_device()
+    if xm.xla_device_hw(device) not in ('TPU', 'CUDA', 'NEURON'):
+      print(f'skip this test for hw {xm.xla_device_hw(device)}')
+      return
+    ordinal = torch.tensor([xr.global_ordinal()],
+                           dtype=torch.float,
+                           device=device)
+    input = ordinal
+    if not use_dynamo:
+      dist.all_reduce(input)
+      xm.mark_step()
+      assert 'xla::AllReduceInPlace' in met.counter_names(
+      ) or 'xla::AllReduce' in met.counter_names()
+    else:
+      compiled_collective = torch.compile(
+          _dynamo_func_wrapper, backend=TestDistCollectiveOpsTpu.my_compiler)
+      compiled_collective(input)
+      torch_xla.sync()
+      assert 'xla::all_reduce' in met.counter_names()
+    return input.cpu()
+
+  @staticmethod
+  def _all_gather_into_tensor(use_dynamo: bool):
```

Comment:
You can simplify this.

First, what we're really testing here is that the same code works for both dynamo and eager, so no need to rewrite the contents of `_dynamo_func_wrapper`. Just give it some generic name, and have a case like this:

```
f = torch.compile(_dist_allgather, ...) if use_dynamo else _dist_allgather
f(input)
torch_xla.sync()
```

You call the same callable either way, and you only conditionally `compile` it.